projects/storefrontlib/src/cms-structure/page/component/handlers/web-component.handler.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  ElementRef,
+  Inject,
+  Injectable,
+  Injector,
+  PLATFORM_ID,
+  Renderer2,
+  ViewContainerRef,
+} from '@angular/core';
+import { Observable } from 'rxjs';
+import { ComponentHandler } from './component-handler';
+import { DOCUMENT, isPlatformBrowser } from '@angular/common';
+import { CxApiService } from '../services/cx-api.service';
+import { CmsComponentMapping, Priority } from '@spartacus/core';
+import { CmsComponentData } from '../../model';
+
+/**
+ * Component handler responsible for launching cms web components.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class WebComponentHandler implements ComponentHandler {
+  constructor(
+    @Inject(DOCUMENT) protected document: any,
+    @Inject(PLATFORM_ID) protected platform: any
+  ) {}
+
+  private loadedWebComponents: { [path: string]: any } = {};
+
+  hasMatch(componentMapping: CmsComponentMapping): boolean {
+    return (
+      typeof componentMapping.component === 'string' &&
+      componentMapping.component.includes('#')
+    );
+  }
+
+  getPriority(): Priority {
+    return Priority.LOW; // low, as it's a default matcher
+  }
+
+  launcher(
+    componentMapping: CmsComponentMapping,
+    viewContainerRef: ViewContainerRef,
+    elementInjector?: Injector
+  ): Observable<[ElementRef]> {
+    return new Observable<[ElementRef]>((observer) => {
+      let webElement;
+      let active = true;
+      const injector = elementInjector ?? viewContainerRef.injector;
+
+      const renderer = injector.get(Renderer2);
+
+      const disposeFunc = () => {
+        active = false;
+        if (webElement) {
+          webElement.remove();
+        }
+      };
+
+      this.initWebComponent(componentMapping.component, renderer).then(
+        (elementName) => {
+          if (elementName) {
+            webElement = renderer.createElement(elementName);
+
+            const cmsComponentData = injector.get(CmsComponentData, null);
+
+            webElement.cxApi = {
+              ...injector.get(CxApiService),
+              cmsComponentData,
+            };
+
+            renderer.appendChild(
+              viewContainerRef.element.nativeElement.parentElement,
```

Comment:
P2: I'm aware it's copy-pasted. But perhaps I spotted a small bug. As far as I understand, this line puts the web component as a **sibling** of the view container ref. But the default component handler seems to put the component **inside** of view container ref. So it looks inconsistent, but maybe I missing something...